react/src/components/calculators/directors/DirectorsForm.tsx

```diff
@@ -66,12 +69,17 @@ export default function DirectorsForm(props: DirectorsFormProps) {
   }
 
   const handleTaxYearChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    invalidateResults()
```

Comment:
this really should be inside if(selectedTaxYear) and preferably after the setTaxYear call - it's only when changes have happened that we want to invalidateResults